### ISSUE TITLE
Data and masks  of GenMOP involved in training

### ISSUE THREAD

**Issue by Yuan-TS** [liuyuan-pal/Gen6D#115]
Dear author, thank you very much for proposing such an easy-to-use and high-performance 6D object pose estimation solution. Now I have some questions to ask:

**First**, taking the chair data of GenMOP as an example, during training or fine-tuning, only chair-ref participates in training, and chair-test does not participate, right?

**Second**, I would like to ask how the Mask of each object in the GenMOP data set is generated?

Looking forward to your reply! Thanks!

**Comment by liuyuan-pal** [comment]
Hi, thank you for your interest in our work!
1. For training objects, we use both ref and test in training. We then evaluate on unseen objects, which are not included in the training set.
2. To generate the mask of the object, I use the reconstructed points of the object and project them onto images to get the object mask. You may also use other foreground mask predictor like `carvekit`.

**Comment by Yuan-TS** [comment]
Thank you very much for your prompt and detailed reply.
 Regarding the problem of mask generation, I found that the **frame*.jpg** in the _images folder_ in the GenMOP data **does not correspond** to the ***.png** in the _mask folder_, such as **frame0.jpg** and **0.png** in knife-ref:
![frame0](https://github.com/liuyuan-pal/Gen6D/assets/34248052/fd9f77a3-ebe7-427b-981f-87dc6ba17ec3)
![0](https://github.com/liuyuan-pal/Gen6D/assets/34248052/8cde3fec-455e-4ba6-93df-36c9c973d3b8)

Will this have an impact on training?

**Comment by liuyuan-pal** [comment]
Yes, there is a map between the COLMAP id and the frame id. You may check https://github.com/liuyuan-pal/Gen6D/blob/50aa71b2dbb3198772c025d54d0dd191a918982c/dataset/database.py#L173

So you can retrieve the input image.

**Comment by Yuan-TS** [comment]
Thanks very much!
I have successfully generated the mask and carried out the training. Both the detector and selector have achieved good results, but the training results of the refiner are a bit less optimistic. Is there any tricks or things that need to be paid attention to?

before finetune
![27-inter](https://github.com/liuyuan-pal/Gen6D/assets/34248052/a089c08e-ef94-4f56-a59f-1d82273ca53c)


after finetune
![27-inter](https://github.com/liuyuan-pal/Gen6D/assets/34248052/b6fa44ac-5b40-4370-b356-16a72fa776e7)

**Comment by ShangQingLiu** [comment]
> Thanks very much! I have successfully generated the mask and carried out the training. Both the detector and selector have achieved good results, but **the training results of the refiner are a bit less optimistic**. Is there any tricks or things that need to be paid attention to?
> 
> before finetune ![27-inter](https://private-user-images.githubusercontent.com/34248052/268543888-a089c08e-ef94-4f56-a59f-1d82273ca53c.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUyNzgyNTIsIm5iZiI6MTcxNTI3Nzk1MiwicGF0aCI6Ii8zNDI0ODA1Mi8yNjg1NDM4ODgtYTA4OWMwOGUtZWY5NC00ZjU2LWE1OWYtMWQ4MjI3M2NhNTNjLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTA5VDE4MDU1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRkMjdmZGNhNDc1NGVjNzczMTBlZTJiZTRhMjU3YmNhMGE5MjBhYzAwMjIyMjkyNmZhYTk4MjY2ZDAzODRiN2UmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.tbfKDcl8w6CJm3lreYdHD_1furuabefnauaF267hosc)
> 
> after finetune ![27-inter](https://private-user-images.githubusercontent.com/34248052/268543911-b6fa44ac-5b40-4370-b356-16a72fa776e7.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUyNzgyNTIsIm5iZiI6MTcxNTI3Nzk1MiwicGF0aCI6Ii8zNDI0ODA1Mi8yNjg1NDM5MTEtYjZmYTQ0YWMtNWI0MC00MzcwLWIzNTYtMTZhNzJmYTc3NmU3LmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTA5VDE4MDU1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkxMzc5MWJjYTlkOGJjMjQ5MjY1NzFjYjUzMTYyMjc4ODY5MGNmYWU4ODA3OTlhNjlkYWZhMjU2MzdiNjMxZDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.d_o8GO9_1EGi1mZMT2gixf37qiZFp8A6vp5U2HREdyo)

Hi , I am interested on your fine-tuning result, is it possible to share the code with me, THANKS!

**Comment by liudifan2** [comment]
> 非常感谢！ 我已经成功生成了掩码并进行了训练。检测器和选择器都取得了不错的效果，但**精炼器的训练效果不太乐观**。有什么技巧或需要注意的地方吗？
> 
> 微调之前 ![27-inter](https://github.com/liuyuan-pal/Gen6D/assets/34248052/a089c08e-ef94-4f56-a59f-1d82273ca53c)
> 
> 微调后 ![27-inter](https://github.com/liuyuan-pal/Gen6D/assets/34248052/b6fa44ac-5b40-4370-b356-16a72fa776e7)

您好题主大大！请问GenMOP数据集中的Mask您是如何得到的？本人QQ邮箱：2498461568@qq.com，微信号：2498461568.如若您能指点一二，我将不胜感激！期待您的回复！

**Comment by liudifan2** [comment]
> > 非常感谢！我已经成功生成了掩码并进行了训练。检测器和选择器都取得了不错的效果，但是**精炼器的训练效果不太乐观**。有什么技巧或者需要注意的地方吗？
> > 微调之前![27-inter](https://github.com/liuyuan-pal/Gen6D/assets/34248052/a089c08e-ef94-4f56-a59f-1d82273ca53c)
> > 微调后![27-inter](https://github.com/liuyuan-pal/Gen6D/assets/34248052/b6fa44ac-5b40-4370-b356-16a72fa776e7)
> 
> 您好题主大大！请问GenMOP数据集中的口罩您是如何得到的？本人QQ邮箱：[2498461568@qq.com](mailto:2498461568@qq.com)，微信号：2498461568。如若能指点一二，我将不胜感激！期待您的回复！

不是口罩，是掩码

**Comment by dewgonlong** [comment]
> > 非常感谢！ 我已经成功生成了掩码并进行了训练。检测器和选择器都取得了不错的效果，但**精炼器的训练效果不太乐观**。有什么技巧或需要注意的地方吗？
> > 微调之前![27-inter](https://github.com/liuyuan-pal/Gen6D/assets/34248052/a089c08e-ef94-4f56-a59f-1d82273ca53c)
> > 微调后![27-inter](https://github.com/liuyuan-pal/Gen6D/assets/34248052/b6fa44ac-5b40-4370-b356-16a72fa776e7)
> 
> 您好题主大大！请问GenMOP数据集中的Mask您是如何得到的？本人QQ邮箱：[2498461568@qq.com](mailto:2498461568@qq.com)，微信号：2498461568.如若您能指点一二，我将不胜感激！期待您的回复！
请问，你知道怎么实现了嘛
